remme/rest_api/openapi.yml

```diff
@@ -306,6 +306,112 @@ paths:
           description: Successfully deleted a key
         404:
           description: No key with the given name found
+  /atomic-swap/init:
+    post:
+      operationId: remme.atomic_swap_tp.rest_api.init
+      summary: Alice may initialize a swap transaction or Bob may follow one as a "non initiator"
+      parameters:
+        - $ref: '#/parameters/swap_init_payload'
+      responses:
+        200:
+          $ref: '#/responses/200BatchSubmitted'
+        500:
+          $ref: '#/responses/500GeneralError'
+
+  /atomic-swap/approve:
+    post:
+      operationId: remme.atomic_swap_tp.rest_api.approve
+      summary: Alice may initialize a swap transaction or Bob may follow one as a "non initiator"
+      parameters:
+        - $ref: '#/parameters/swap_approve_payload'
+      responses:
+        200:
+          $ref: '#/responses/200BatchSubmitted'
+        500:
+          $ref: '#/responses/500GeneralError'
+
+  /atomic-swap/expire:
+    post:
+      operationId: remme.atomic_swap_tp.rest_api.expire
+      summary: Alice or Bob expire the swap and receive the refund
+      parameters:
+        - $ref: '#/parameters/swap_approve_payload'
+      responses:
+        200:
+          $ref: '#/responses/200BatchSubmitted'
+        500:
+          $ref: '#/responses/500GeneralError'
+  /atomic-swap/set-secret-lock:
+    post:
+      operationId: remme.atomic_swap_tp.rest_api.set_secret_lock
+      summary: Bob may set the secret lock
+      parameters:
+        - $ref: '#/parameters/swap_set_secret_lock_payload'
+      responses:
+        200:
+          $ref: '#/responses/200BatchSubmitted'
+        500:
+          $ref: '#/responses/500GeneralError'
+  /atomic-swap/close:
+    post:
+      operationId: remme.atomic_swap_tp.rest_api.close
+      summary: Close the swap by providing a secret key
+      parameters:
+        - $ref: '#/parameters/swap_close_payload'
+      responses:
+        200:
+          $ref: '#/responses/200BatchSubmitted'
+        500:
+          $ref: '#/responses/500GeneralError'
+  /atomic-swap/{swap_id}:
+    get:
+      operationId: remme.atomic_swap_tp.rest_api.get_swap_info
+      summary: Get information about swap
+      parameters:
+        - name: swap_id
+          description: The id of the batch to check
+          in: path
+          type: string
+          pattern: '^[0-9a-f]{64}$'
+          required: true
+      responses:
+        200:
+          $ref: '#/responses/SwapInfo'
+        404:
+          description: No batch with the given id found
+        500:
+          $ref: '#/responses/500GeneralError'
+
+  /utils/generate-random-key256:
```

Comment:
Do we need utils in API ? 
maybe rest api clients should do that on their side ?

Reply:
it is used for the unification of cryptographic tool for consistency if a client wants to generate keys remotely, he may reference the very same source to match requirements

Reply:
Don't think we need it in REST API. It is not something that cannot be easily done on the client side.